### ISSUE TITLE
Unable to use custom_script attribute to run a script when a windows vm is being spun up

### ISSUE THREAD

**Issue by Divya1388** [hashicorp/packer-plugin-azure#260]
#### Overview of the Issue

Trying to run a user data script using the attribute custom_script released last week and getting the following error

packer build novalidhostname.pkr.hcl                                          Error: Unsupported argument                                                                     

  on novalidhostname.pkr.hcl line 56:
  (source code not available)

An argument named "custom_script" is not expected here.

### Plugin and Packer version

From `packer version` 1.8.3

### Simplified Packer Buildfile

packer {
  required_plugins {
    azure = {
      version = ">= 1.3.1"
      source  = "github.com/hashicorp/azure"
    }
  }
}

source "azure-arm" "windows" {
  client_id       = var.client_id
  client_secret   = var.client_secret
  subscription_id = var.subscription_id
  tenant_id       = var.tenant_id

  shared_image_gallery {
    subscription   = "fe76ae45-ee1b-4888-a44e-*********"
    resource_group = "image-build-rg"
    gallery_name   = "Published"
    image_name     = Win2019"
    image_version  = "latest"
  }
  managed_image_name                  = "custompkrimage1"
  managed_image_resource_group_name   = "*************"
  location                            = "centralus"
  vm_size                             = "Standard_D2_v2"
  virtual_network_name                = "********************"
  virtual_network_subnet_name         = "development"
  virtual_network_resource_group_name = "*******************"
  os_type                             = "Windows"
  communicator                        = "winrm"
  winrm_insecure                      = "true"
  winrm_timeout                       = "3m"
  winrm_use_ssl                       = "true"
  winrm_username                      = "packer"
  custom_script   = "powershell -ExecutionPolicy Unrestricted -NoProfile -NonInteractive -Command \"$userData = (Invoke-RestMethod -Headers @{Metadata=$true} -Method GET -Uri http://169.254.169.254/metadata/instance/compute/userData?api-version=2021-01-01$([char]38)format=text); $contents = [System.Text.Encoding]::UTF8.GetString([Convert]::FromBase64String($userData)); set-content -path c:\\Windows\\Temp\\userdata.ps1 -value $contents; . c:\\Windows\\Temp\\userdata.ps1;\""
  user_data_file  = "./scripts/userdata.ps1"
}

build {
  sources = ["sources.azure-arm.windows"]

  provisioner "powershell" {
    inline = ["New-Item -Path C:\\packages -ItemType Directory"]
  }
}

### Operating system and Environment details

Windows 2019

### Log Fragments and crash.log files

  on novalidhostname.pkr.hcl line 56:
  (source code not available)

An argument named "custom_script" is not expected here.

Set the env var `PACKER_LOG=1` for maximum log detail.

**Comment by matt-richardson** [comment]
This hasn't been released yet. 
The last release was ~5 months ago. 

I believe that @JenGoldstrich is planning on cutting a release soon.

**Comment by Divya1388** [comment]
@matt-richardson  Thanks for letting me know. 
@JenGoldstrich  Thanks for your effort on this. Do we have an ETA for this to be released?

**Comment by JenGoldstrich** [comment]
Hi all! 

I have opened a PR here https://github.com/hashicorp/packer-plugin-azure/pull/259 to enable disk encryption with CMKs on Azure, and am planning to release once I have finished the tests, and gotten approval from another member of the team on that code change. This should happen within 1-2 weeks

I'm glad we have users that are excited to use the custom script that you developed @matt-richardson, and am eager to get my PR done so we can release it for everyone!
